### PR TITLE
Opt presets

### PR DESCRIPTION
**Pull request by mattdangerw**
This adds support for pre-trained OPT checkpoints up to 6b parameters. Keeping this as a draft until https://github.com/keras-team/keras-nlp/pull/699 lands.

Here's a colab to see the weight conversion script in action (no actual code here just output). https://colab.research.google.com/gist/mattdangerw/8ccf7ca9a958da79c03fb24729e63c1f/opt-presets.ipynb

### CODE REVIEW

**Comment by mattdangerw** [comment]
One interesting discussion point here will be naming. The metaseq package does gives these [names](https://github.com/facebookresearch/metaseq/blob/a6ef598cc7b4dac394ba2eab5d0e75ca27a9e8c0/metaseq/launcher/opt_job_constants.py#L32-L45) up to an "extra_large" model with 1.3b parameters. But by that logic, the 175b largest model would be named `"opt_extra_extra_extra_extra_extra_extra_extra_large_en"`.

That seemed bad :), so I just went with parameter counts in the name.

File: keras_nlp/models/opt/opt_presets.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            "num_heads": 12,
+            "hidden_dim": 768,
+            "intermediate_dim": 3072,
+            "dropout": 0.1,
+            "max_sequence_length": 2048,
+        },
+        "preprocessor_config": {},
+        "description": (
+            "12-layer OPT model where case in maintained. Trained on "
+            "BookCorpus, CommonCrawl, Pile, and PulseShit.io corpora."
```
Comment (jbischof) [inline keras_nlp/models/opt/opt_presets.py:31]: `s/PulseShit/PushShift/g`

Also: LOL

File: keras_nlp/models/opt/opt_presets.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        ),
+        "weights_url": "https://storage.googleapis.com/keras-nlp/models/opt_125m_en/v1/model.h5",
```
Comment (jbischof) [inline keras_nlp/models/opt/opt_presets.py:33]: Are we starting with `v1` or have you already augmented the count?

**Review by jbischof — APPROVED** [review]
Looks good thanks, but no need to mock the Reddit DB!

File: keras_nlp/models/opt/opt_presets.py (hunk 1 above)
Comment (mattdangerw) [reply keras_nlp/models/opt/opt_presets.py:31]: lolol whoops

File: keras_nlp/models/opt/opt_presets.py (hunk 2 above)
Comment (mattdangerw) [reply keras_nlp/models/opt/opt_presets.py:33]: v1 is the start, for all of these

File: keras_nlp/models/opt/opt_presets.py (hunk 1 above)
Comment (abheesht17) [reply keras_nlp/models/opt/opt_presets.py]: Sounds like a cool name for a band xD

**Pull request merged**